pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1687,70 +1690,49 @@
     ::  +sign-to-channel-event: strip the vase from a sign:agent:gall
     ::
     ++  sign-to-channel-event
-      |=  =sign:agent:gall
-      ^-  channel-event
-      ?.  ?=(%fact -.sign)  sign
-      [%fact [p q.q]:cage.sign]
+      |=  [=sign:agent:gall =channel request-id=@ud]
+      ^-  (unit channel-event)
+      ?.  ?=(%fact -.sign)  `sign
+      ?~  desk=(app-to-desk channel request-id)  ~
+      :-  ~
+      [%fact u.desk [p q.q]:cage.sign]
     ::  +app-to-desk
     ::
     ++  app-to-desk
       |=  [=channel request-id=@ud]
       ^-  (unit desk)
       =/  sub  (~(get by subscriptions.channel) request-id)
       ?~  sub
-        ((trace 0 |.("no subscription for request-id {(trip request-id)}")) ~)
+        ((trace 0 |.("no subscription for request-id {<request-id>}")) ~)
       =/  des=(unit (unit cage))
         (rof ~ %gd [our app.u.sub da+now] ~)
       ?.  ?=([~ ~ *] des)
         ((trace 0 |.("no desk for app {<app.u.sub>}")) ~)
       `!<(=desk q.u.u.des)
-    ::  +channel-event-to-sign: attempt to recover a sign from a channel-event
+    ::  +channel-event-to-json: render channel event as json channel event
     ::
-    ++  channel-event-to-sign
-      ~%  %eyre-channel-event-to-sign  ..part  ~
+    ++  channel-event-to-json
+      ~%  %eyre-channel-event-to-json  ..part  ~
       |=  [=channel request-id=@ud event=channel-event]
-      ^-  (unit sign:agent:gall)
-      ?.  ?=(%fact -.event)  `event
-      ::  rebuild vase for fact data
-      ::
-      =/  des=(unit desk)  (app-to-desk channel request-id)
-      ?~  des  ~
-      =*  have=mark  mark.event
-      =/  val=(unit (unit cage))
-        (rof ~ %cb [our u.des da+now] /[have])
-      ?.  ?=([~ ~ *] val)
-        ((trace 0 |.("no mark {(trip have)}")) ~)
-      =+  !<(=dais:clay q.u.u.val)
-      =/  res  (mule |.((vale:dais noun.event)))
-      ?:  ?=(%| -.res)
-        ((trace 0 |.("stale fact of mark {(trip have)}")) ~)
-      `[%fact have p.res]
-    ::  +sign-to-json: render sign from request-id as json channel event
-    ::
-    ++  sign-to-json
-      ~%  %sign-to-json  ..part  ~
-      |=  [=channel request-id=@ud =sign:agent:gall]
       ^-  (unit (quip move json))
       ::  for facts, we try to convert the result to json
       ::
       =/  [from=(unit [=desk =mark]) jsyn=(unit sign:agent:gall)]
-        ?.  ?=(%fact -.sign)       [~ `sign]
-        ?:  ?=(%json p.cage.sign)  [~ `sign]
+        ?.  ?=(%fact -.event)       [~ `event]
+        ?:  ?=(%json mark.event)    [~ `[%fact %json !>((json noun.event))]]
```

Comment:
Doing `(json noun.event)` is a little bit scary here. Is it possible that we end up with a non-json noun in this code path? Not sure how and where the `sign:agent:gall` is verified.

Reply:
Unitized this too because of out of band conversation. Avoiding crashing in eyre is good because it lessens the chance of poorly written userspace applications bricking your ship.